services/wireguard/network/network_linux.go

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright (C) 2018 The "MysteriumNetwork/node" Authors.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package network
+
+import (
+	"fmt"
+	"net"
+	"os/exec"
+
+	"github.com/mdlayher/wireguardctrl"
+	"github.com/mdlayher/wireguardctrl/wgtypes"
+)
+
+// NewNetwork creates Wireguard client with predefined interface name and public IP.
+func NewNetwork(name, publicIP string) (*network, error) {
+	wgClient, err := wireguardctrl.New()
+	if err != nil {
+		return nil, err
+	}
+
+	return &network{
+		name:     name,
+		wgClient: wgClient,
+	}, nil
+}
+
+// Provider starts and configure wireguard network interface for providing service.
+// It returns information required to establish connection to the service.
+func (n *network) Provider() (Provider, error) {
+	if _, err := n.wgClient.Device(n.name); err != nil {
+		if err := exec.Command("ip", "link", "add", "dev", n.name, "type", "wireguard").Run(); err != nil {
+			return Provider{}, err
+		}
+	}
+
+	if err := exec.Command("ip", "address", "replace", "dev", n.name, "192.168.100.1/24").Run(); err != nil {
```

Comment:
I would suggest not to use 192.168.x.x, since it has higher probability of clash. Lets use some random from 10.x.x.x as per https://tools.ietf.org/html/rfc1918

```suggestion
	if err := exec.Command("ip", "address", "replace", "dev", n.name, "10.182.47.0/24").Run(); err != nil {
```

Reply:
Changed it to the mocked network from 10.0.0.0/8.